specification.md

```diff
@@ -1357,44 +1359,86 @@ For more information about workflow expressions, reference the [Workflow Express
+Note, that without the closing bracket at the very end of the string, it won't 
+be evaluated as an expression.
+
+A workflow expression can also be defined in an [expression function definition](#Using-Functions-For-Expression-Evaluation) and then applied by reference.
+To reference a expression function, an _expression reference string_ is used that 
+starts with the `$fn{` token and ends with `}`.
+The _expression reference string_ has the format `$fn{<function name>}`, where 
+`<function name>` has to match the user-specified `name` of the defined 
+expression function of `type` `"expression"`.
+
+On the rare occasion that one needs to create a string that literally looks like 
+an _expression string_ or an _expression reference string_, the dollar sign 
+should be repeated. Only if the remainder matches a workflow expression markup,
+the runtime reduces the double dollar signs `$$` to a single dollar sign `$` and 
+refrains from evaluating it as a workflow expression.
 
-Note that using a non-default expression language could lower the portability of your workflow definitions 
-across multiple container/cloud platforms.
+```text
+"foo": "$${ this is text in a string starting with '${' and ending with a single '}' }"
+"foo": "$$fn{ this is text in a string starting with '$fn{' and ending with a single '}' }"
+```
+
+The expression language gives access to the state data to select or transform 
+the state data when processing the workflow expression. It also provides access 
+to [workflow secrets](#Workflow-secrets) and 
+[workflow constants](#Workflow-constants).
+
+Note that different data filters play a big role as to which parts of the 
+states data are to be used when the expression is evaluated during workflow
+execution, e.g. a [State Data Filter](#State-data-filters) `input`
+expression can only access data that has been passed to that state, inside an action, 
+only the data filtered by the [Action Data Filter](#Action-data-filters) `fromStateData` 
+parameter are available to generate arguments to an action, etc.
+Please refer to the specification of filters for more inforamtion, 
+e.g. [State Data Filter](#State-data-filters), [Action Data Filter](#Action-data-filters), 
+[Event Data Filter](#Event-data-filters) and see the specification of individual 
+parameters if you want to set them using an expression.
+
+The default expression language is [jq](https://stedolan.github.io/jq/).
+This version of the workflow language supports jq [version 1.6](https://github.com/stedolan/jq/releases/tag/jq-1.6) syntax.
+You can find more information on jq v1.6 in its [manual](https://stedolan.github.io/jq/manual/v1.6/).
+
+JQ expressions provide the state data as the root document (`.`). 
+Global [secrets](#Workflow-secrets) and [constants](#Workflow-constants) can be 
+accessed through the jq variables `$SECRETS` and `$CONST` respectively, which 
+makes them reserved variable names in JQ expressions.
```

Comment:
There's probably a chance of wanting to expose _other_ data in the future as well, so it would make more sense to reserve either a pattern (e.g., a leading underscore with `$_SECRETS` and `$_CONST` being the first two uses) or a single object-valued variable (e.g., `$SPECIAL` with `SECRETS` and `CONST` keys).

Reply:
I think I would agree but I have to defer this proposal to become its own PR.